src/Period.php

```diff
@@ -137,6 +137,10 @@ public static function createFromDay($day)
     {
         $startDate = static::filterDatePoint($day)->setTime(0, 0, 0);
 
+        if ($startDate->format('u') !== '000000') {
+            $startDate = new DateTimeImmutable($startDate->format(static::DATE_ISO8601));
```

Comment:
If we use the Period::DATE_ISO8601 constant, we will loose PHP DateTimeZone Info. Maybe it's better to use 
```php
 $startDate = new DateTimeImmutable(
   $startDate->format('Y-m-d H:i:s.000000'),
   $startDate->getTimezone()
);
```
If a little more verbose but less error prone and preserve the DateTimeZone information. You should add tests to validate my assumption too.